tasktiger/__init__.py

```diff
@@ -89,8 +89,12 @@ def __init__(self, connection=None, config=None, setup_structlog=False):
             'DEFAULT_QUEUE': 'default',
 
             # After how many seconds time out on listening on the activity
-            # channel and check for scheduled or expired items.
+            # channel and check for scheduled or expired items.  The batch
+            # timeout will delay the specified seconds after the first message
+            # to wait for additional messages, useful for very active systems.
+            # Appropriate values: 0 <= SELECT_BATCH_TIMEOUT <= SELECT_TIMEOUT
             'SELECT_TIMEOUT': 1,
+            'SELECT_BATCH_TIMEOUT': 0,
```

Comment:
This deals with the specific use case where you have many workers processing a very active queue.  Assume we start with an empty queue. If you have 20 workers and put 1 task on the queue all 20 workers will detect this as a `new_queue_found` and immediately exit the `_wait_for_new_tasks` loop.  1 worker will successfully move that task from `QUEUED` to `ACTIVE` and 19 workers will fail to move any tasks because `QUEUED` is now empty.  In this very simple case we wasted 19 redis calls.

Now in an active queue where we are constantly queueing tasks, by waiting just a little before exiting the `_wait_for_new_tasks` we let `QUEUED` to fill with a few more tasks so that more that 1 out of the 20 will have a task available to move from `QUEUED` to `ACTIVE`.  Maybe something like `SELECT_FILL_DELAY` would be a better name than `SELECT_BATCH_TIMEOUT`.

I'm not exactly sure what you mean by using a `sleep`.  It sleeps in the `get_message` so that we always get messages off the channel if they are available.  A regular sleep would have the same result of letting `QUEUED` to fill but maybe leaving a few messages on the channel that would be a false positive the next time we entered the loop.